astropy/table/operations.py

```diff
@@ -380,6 +380,68 @@ def unique(input_table, keys=None, silent=False):
     return unique_table
 
 
+def remove_duplicate_rows(input_table, keys=None, keep='first', silent=False):
+    """Remove duplicate rows from a table
+
+    Parameters
+    ----------
+    input_table : `~astropy.table.Table` object or a value that
+        will initialize an `~astropy.table.Table` object
+
+    keys : str or list of str
+        Name(s) of column(s) used to unique rows.
+        Default is to use all columns.
+
+    keep : one of 'first', 'last' or False
+        Whether to keep the first or last row for each set of
+        duplicates. If False, all rows in the duplicate set are
+        removed.
+
+    silent : boolean, default `False`
+        If `True` masked value column(s) are silently removed from
+        ``keys``. If `False` an exception is raised when ``keys`` contains
+        masked value column(s).
+
+    Returns
+    -------
+        A copy of the table with duplicate rows removed
+
+    """
+
+    if keep not in ('first', 'last', False):
+        raise ValueError("'keep' should be one of 'first', 'last', False")
+
+    if keys is None:
+        keys = input_table.colnames
+
+    if input_table.masked:
+        if isinstance(keys, six.string_types):
+            keys = [keys]
+        nkeys = 0
+        for i, key in enumerate(keys[:]):
+            if np.any(input_table[key].mask):
+                if not silent:
+                    raise ValueError("cannot unique masked value key columns, "
+                                     "remove column '{0}' from keys and rerun "
+                                     "remove_duplicate_rows".format(key))
+                del keys[i]
+        if len(keys) == 0:
+            raise ValueError("no column remained in ``keys``, "
+                             "remove_duplicate_rows cannot "
+                             "work with masked value key columns")
+
+    grouped_table = input_table.group_by(keys)
+    if keep == 'first':
+        cleaned_table = grouped_table[grouped_table.groups.indices[:-1]]
+    elif keep == 'last':
+        cleaned_table = grouped_table[grouped_table.groups.indices[1:]-1]
+    else:
+        cleaned_table = vstack([t for t in grouped_table.groups
```

Comment:
Done. I tend to get confused when doing tricks with `np.diff` and boolean masks (since `np.diff` loses one element), so I hope I covered all edge cases in the tests. And indeed, a pretty major speed increase this way.